internal/controllers/clusterextension_controller.go

```diff
@@ -97,12 +140,36 @@ func (r *ClusterExtensionReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	}
 
 	if updateFinalizers {
-		if updateErr := r.Update(ctx, reconciledExt); updateErr != nil {
-			return res, utilerrors.NewAggregate([]error{reconcileErr, updateErr})
+		if updateErr := r.Client.Update(ctx, reconciledExt); updateErr != nil {
+			updateErrors = append(updateErrors, updateErr)
 		}
 	}
 
-	return res, reconcileErr
+	if reconcileErr != nil {
+		updateErrors = append(updateErrors, reconcileErr)
+	}
+
+	return res, utilerrors.NewAggregate(updateErrors)
```

Comment:
Optional Nit: Go's std lib `errors` package now has an [`errors.Join()`](https://github.com/operator-framework/helm-operator-plugins/pull/337) function that joins errors into a single error similar to this Aggregate error from the k8s util/errors package. Where possible I'd personally prefer we use the std lib than another library (even if it is a k8s library)

Reply:
Looking at the `errors.Join()` API, it does not use error aggregation (maybe not a big deal), but instead specifically states "wrap", but not can not be unwrapped via `errors.Unwrap()`? It also doesn't accept an array of `error`s, but only a list, so it's actually not very flexible... I think you'd actually have to call `errors.Join()` at every point.

Reply:
To clarify, errors returned by `errors.Join()` have an `Unwrap() []error` function, not `Unwrap() error`‽‽

Reply:
>specifically states "wrap", but can not be unwrapped via errors.Unwrap()?

Correct, it has it's own unwrap method that returns the slice of errors.

>It also doesn't accept an array of errors, but only a list, so it's actually not very flexible... I think you'd actually have to call errors.Join() at every point.

In this case you should be able to do:
```go
return res, errors.Join(updateErrors...)
```

---

Totally optional nit to address and I am fine with it staying as an Aggregate error. Just thought that it might be nice to suggest using the new function in the standard library :)

Reply:
We'll handle errors in #878.